oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/tree/impl/AbstractTree.java

```diff
@@ -150,34 +151,59 @@ protected Iterable<String> getChildNames() {
 
     @Override
     public String toString() {
-        return toString(5);
+        return toJsonString(2);
     }
 
-    private String toString(int childNameCountLimit) {
-        StringBuilder sb = new StringBuilder();
-        sb.append(getPath()).append(": ");
-
-        sb.append('{');
-        for (PropertyState p : getProperties()) {
-            sb.append(' ').append(p).append(',');
-        }
-
-        Iterator<String> names = this.getChildNames().iterator();
-        int count = 0;
-        while (names.hasNext() && ++count <= childNameCountLimit) {
-            sb.append(' ').append(names.next()).append(" = { ... },");
+    /**
+     * Represents an OAK tree as a JSON object string
+     * When the representation is limited in depth, {"...":"..."} represents the limit.
+     * Note that, try-catch sentences are used to add "ERROR: "+e.getMessage() values in case of representation error.
+     * Note that, this is a recursive method.
+     * @param depth tree depth to represent. Use -1 for an unlimited depth.
+     * @return json object representation of the tree as a string
+     */
+    public String toJsonString(int depth) {
+        if (depth == 0) {
+            return quote("...");
         }
-
-        if (names.hasNext()) {
-            sb.append(" ...");
+        StringBuilder str = new StringBuilder();
+        str.append("{");
+        str.append(quote("_properties_") + ":{ ");
+        for (PropertyState ps : this.getProperties()) {
+            str.append(quote(ps.getName()) + ":");
```

Comment:
To replace the trailing comma with a prefix comma:

    int count = 0;
    for (PropertyState ps : this.getProperties()) {
        if (count++ > 0) {
            buff.append(", ");
        }

Reply:
or

~~~
String sep = "";
for (PropertyState ps : this.getProperties()) {
    buf.append(sep);
    sep = ", ";
~~~